manager/ui/war/plugins/api-manager/ts/directives.ts

```diff
@@ -636,4 +611,205 @@ module Apiman {
+        };
+    });
+
+    _module.directive('regexpValidation', function () {
+        return {
+            require: 'ngModel',
+            link: function (scope, element, attributes, ngModel: any) {
+                ngModel.$parsers.push(function (value?: string) {
+                    if (value && value.length !== 0) {
+                        try {
+                            var regexp = new RegExp(value)
+                            ngModel.$setValidity('text', true)
+                        } catch(error) {
+                            ngModel.$setValidity('text', false)
+                        }
+                    } else {
+                        ngModel.$setValidity('text', true)
+                    }
+
+                    return value;
+                });
+            }
+        };
+    });
+
+    _module.directive('jsonValidation', function () {
+        return {
+            require: 'ngModel',
+            link: function (scope, element, attributes, ngModel: any) {
+                ngModel.$parsers.push(function (value?: string) {
+                    if (value && value.length !== 0) {
+                        try {
+                            var json = JSON.parse(value)
+                            ngModel.$setValidity('json', true)
+                        } catch(error) {
+                            ngModel.$setValidity('json', false)
+                        }
+                    } else {
+                        ngModel.$setValidity('json', true)
+                    }
+
+                    return value;
+                });
+            }
+        };
+    });
+
+    _module.directive('checkGatewayPassword', function () {
+        return {
+            require: 'ngModel',
+            link: function (scope, element, attributes, ngModel: any) {
+                ngModel.$parsers.push(function (value?: string) {
+                    if (scope.gatewayForm.password.$viewValue !== scope.gatewayForm.passwordConfirm.$viewValue) {
+                        scope.gatewayForm.password.$setValidity('password', false)
+                        scope.gatewayForm.passwordConfirm.$setValidity('password', false)
+                    } else {
+                        scope.gatewayForm.password.$setValidity('password', true)
+                        scope.gatewayForm.passwordConfirm.$setValidity('password', true)
+                    }
+
+                    return value;
+                });
+            }
+        };
+    });
+
+    _module.directive('checkGatewayEndpoint', function () {
+        return {
+            require: 'ngModel',
+            link: function (scope, element, attributes, ngModel: any) {
+                ngModel.$parsers.push(function (value?: string) {
+                    if (value && value.length !== 0) {
+                        var GATEWAY_VALIDATION = /^\$\{apiman\.gateway-endpoint:https?:\/\/[A-z0-9_\-\.]+(:[0-9]+)?(\/apiman-gateway-api)?\}$|^https?:\/\/[A-z0-9_\-\.]+(:[0-9]+)?(\/apiman-gateway-api)?$/

```

Comment:
Here we must support http:

- the default configuration is http
- in Docker it is very common to use http for the internal communication

Reply:
The "?" after the "s" in "https" makes the "s" optional.